### PR TITLE
8345004: [BACKOUT] GTK & Nimbus LAF: Tabbed pane's background color is not expected one when change the opaque checkbox.

### PR DESCRIPTION
**Pull request by kumarabhi006**
The fix for [JDK-8226990](https://bugs.openjdk.org/browse/JDK-8226990) causes regression in dark theme for GTK L&F on Ubuntu system and that results in manual test failures identified recently.

Few of the changes made in [JDK-8226990](https://bugs.openjdk.org/browse/JDK-8226990) is further modified in [JDK-8335130](https://bugs.openjdk.org/browse/JDK-8335130).

Backing out the overall fix to prevent any regression.

CI testing after revert is fine.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found trailing period in issue title for `8345004: [BACKOUT] GTK & Nimbus LAF: Tabbed pane's background color is not expected one when change the opaque checkbox.`

### Issue
 * [JDK-8345004](https://bugs.openjdk.org/browse/JDK-8345004): [BACKOUT] GTK &amp; Nimbus LAF: Tabbed pane's background color is not expected one when change the opaque checkbox. (**Bug** - P2)

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**)
 * [Alexander Zuev](https://openjdk.org/census#kizune) (@azuev-java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22379/head:pull/22379` \
`$ git checkout pull/22379`

Update a local copy of the PR: \
`$ git checkout pull/22379` \
`$ git pull https://git.openjdk.org/jdk.git pull/22379/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22379`

View PR using the GUI difftool: \
`$ git pr show -t 22379`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22379.diff">https://git.openjdk.org/jdk/pull/22379.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22379#issuecomment-2499694213)
</details>

### CODE REVIEW

**Comment by prsadhuk** [comment]
guess you should be using [JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) to backout the fix..

**Comment by kumarabhi006** [comment]
> guess you should be using [JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) to backout the fix..

[JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) is marked as confidential and the initial fix was in open repo, thought of creating a new bug for backout. Once it is done.. will close [JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) this issue.

**Comment by azuev-java** [comment]
There was a test in the original PR. test/jdk/javax/swing/JTabbedPane/TestJTabbedPaneOpaqueColor.java - should it be corrected or problem-listed if you back out all the changes in the fix? Or it will remain valid even without the original fix?

**Comment by prsadhuk** [comment]
> > guess you should be using [JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) to backout the fix..
> 
> [JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) is marked as confidential and the initial fix was in open repo, thought of creating a new bug for backout. Once it is done.. will close [JDK-8340998](https://bugs.openjdk.org/browse/JDK-8340998) this issue.

Although I dont think there is anything there to be confidential now and you can mark it non-confidential, I am ok with that being closed as dup of this..Please submit a "Redo" bug for [JDK-8226990](https://bugs.openjdk.org/browse/JDK-8226990)

**Comment by prsadhuk** [comment]
> > There was a test in the original PR. test/jdk/javax/swing/JTabbedPane/TestJTabbedPaneOpaqueColor.java - should it be corrected or problem-listed if you back out all the changes in the fix? Or it will remain valid even without the original fix?
> 
> Being a manual test, don't think it needs to be problem-listed. This test is incorrect after backing out the changes. Once this backout is integrated, will open a new bug to address the original issue.
> 
> Forgot to mention about the test in JBS, will add this now.

I guess @azuev-java is right...TestJTabbedPaneOpaqueColor.java was formed from an existing test which was problemlisted and I removed it later on [JDK-8339577](https://bugs.openjdk.org/browse/JDK-8339577) so you can problemlist this test now under "# Client manual tests" section so that you dont forget and tag it to that "Redo" JBS issue..

**Comment by kumarabhi006** [comment]
> Although I dont think there is anything there to be confidential now and you can mark it non-confidential, I am ok with that being closed as dup of this..Please submit a "Redo" bug for [JDK-8226990](https://bugs.openjdk.org/browse/JDK-8226990)

Filed a redo bug [JDK-8345089](https://bugs.openjdk.org/browse/JDK-8345089)

**Comment by prsadhuk** [comment]
> > Although I dont think there is anything there to be confidential now and you can mark it non-confidential, I am ok with that being closed as dup of this..Please submit a "Redo" bug for [JDK-8226990](https://bugs.openjdk.org/browse/JDK-8226990)
> 
> Filed a redo bug [JDK-8345089](https://bugs.openjdk.org/browse/JDK-8345089)

Seems there is a format for this backout/redo and yours is not in accordance...See https://openjdk.org/guide/#backing-out-a-change, 
original should be cloned and new JBS should be in form of "REDO: <old bug description>"

**Comment by kumarabhi006** [comment]
> I guess @azuev-java is right...TestJTabbedPaneOpaqueColor.java was formed from an existing test which was problemlisted and I removed it later on [JDK-8339577](https://bugs.openjdk.org/browse/JDK-8339577) so you can problemlist this test now under "# Client manual tests" section so that you dont forget and tag it to that "Redo" JBS issue..

Test you have removed was problem-listed for `macOS` and as per [JDK-8331619](https://bugs.openjdk.org/browse/JDK-8331619) test is no longer valid for `Aqua L&F`. `TestJTabbedPaneOpaqueColor.java` is restricted to run only on **Linux and Windows** platform.
Do we still need to problemlist it?

**Comment by kumarabhi006** [comment]
> Seems there is a format for this backout/redo and yours is not in accordance...See https://openjdk.org/guide/#backing-out-a-change, original should be cloned and new JBS should be in form of "REDO: [old bug description]"

Do I need to close the REDO bug [JDK-8345089](https://bugs.openjdk.org/browse/JDK-8345089) and create a new one cloned with original issue now?
What about the `Backout` issue?

File: src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java [hunk 1]
```diff
@@ -153,9 +146,6 @@ private boolean scrollableTabLayoutEnabled() {
      */
     @Override
     protected void installDefaults() {
-        selectColor = UIManager.getColor("TabbedPane.selected");
-        contentOpaque = UIManager.getBoolean("TabbedPane.contentOpaque");
-        unselectedBackground = UIManager.getColor("TabbedPane.unselectedBackground");
         updateStyle(tabPane);
```
Comment (prsadhuk) [inline src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java:149]: guess you missed adding this back
tabsOpaque = UIManager.getBoolean("TabbedPane.tabsOpaque");

File: src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java [hunk 2]
```diff
@@ -647,12 +637,6 @@ private void paintTab(SynthContext ss, Graphics g,
             }
         }
 
-        if (isSelected) {
-            g.setColor(selectColor);
-        } else {
-            g.setColor(getUnselectedBackgroundAt(tabIndex));
-        }
-
         tabContext.getPainter().paintTabbedPaneTabBackground(tabContext, g,
```
Comment (prsadhuk) [inline src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java:640]: and this
if (tabsOpaque || tabPane.isOpaque()) {

File: src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java (hunk 1 above)
Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java:149]: ok..I guess I am wrong..

Comment (kumarabhi006) [reply src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java:149]: Not actually... `tabsOpaque` property was added in JDK-8226990 but later on it it is moved to SynthPainterImpl and GTKPainter classes under [JDK-8335130](https://bugs.openjdk.org/browse/JDK-8335130) issue.

File: src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java (hunk 2 above)
Comment (kumarabhi006) [reply src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTabbedPaneUI.java:640]: same reason as above.

**Comment by prsadhuk** [comment]
> > I guess @azuev-java is right...TestJTabbedPaneOpaqueColor.java was formed from an existing test which was problemlisted and I removed it later on [JDK-8339577](https://bugs.openjdk.org/browse/JDK-8339577) so you can problemlist this test now under "# Client manual tests" section so that you dont forget and tag it to that "Redo" JBS issue..
> 
> Test you have removed was problem-listed for `macOS` and as per [JDK-8331619](https://bugs.openjdk.org/browse/JDK-8331619) test is no longer valid for `Aqua L&F`. `TestJTabbedPaneOpaqueColor.java` is restricted to run only on **Linux and Windows** platform. Do we still need to problemlist it?

This test will now fail in windows and linux, right? In that case, you need to PL it for this till you fix the REDO bug..

**Comment by prsadhuk** [comment]
> > Seems there is a format for this backout/redo and yours is not in accordance...See https://openjdk.org/guide/#backing-out-a-change, original should be cloned and new JBS should be in form of "REDO: [old bug description]"
> 
> Do I need to close the REDO bug [JDK-8345089](https://bugs.openjdk.org/browse/JDK-8345089) and create a new one cloned with original issue now? What about the `Backout` issue?

Better to follow the process and do the steps listed...

**Comment by kumarabhi006** [comment]
> This test will now fail in windows and linux, right? In that case, you need to PL it for this till you fix the REDO bug..

Yes, it will fail on windows and linux. I will add it the PL.

**Comment by kumarabhi006** [comment]
> This test will now fail in windows and linux, right? In that case, you need to PL it for this till you fix the REDO bug..

Updated the PL.

**Comment by kumarabhi006** [comment]
@prsadhuk Updated the JBS title of [JDK-8345004](https://bugs.openjdk.org/browse/JDK-8345004) to reflect it as a backout issue. Only thing missing from the process is it should be craeted as a "sub-task" of REDO bug but I guess this should be ok as the bugs are linked properly.

**Comment by prsadhuk** [comment]
> > > > Seems there is a format for this backout/redo and yours is not in accordance...See https://openjdk.org/guide/#backing-out-a-change, original should be cloned and new JBS should be in form of "REDO: [old bug description]"
> > > 
> > > 
> > > Do I need to close the REDO bug [JDK-8345089](https://bugs.openjdk.org/browse/JDK-8345089) and create a new one cloned with original issue now? What about the `Backout` issue?
> > 
> > 
> > Better to follow the process and do the steps listed...
> 
> Cloned the Original issue to create a Redo [JDK-8345090](https://bugs.openjdk.org/browse/JDK-8345090) bug. Closed [JDK-8345089](https://bugs.openjdk.org/browse/JDK-8345089) Redo bug as duplicate.

Fixed Version should be "25" right? Also, SUbtask "[CLONE - Deproblemlist javax/swing/JTabbedPane/4690946/bug4690946.java](https://bugs.openjdk.org/browse/JDK-8345091)" should be closed..

**Review by prsadhuk — APPROVED** [review]

**Comment by kumarabhi006** [comment]
Fixed Version should be "25" right? Also, SUbtask "[CLONE - Deproblemlist javax/swing/JTabbedPane/4690946/bug4690946.java](https://bugs.openjdk.org/browse/JDK-8345091)" should be closed..

Yes... fixed version for Redo is updated to 25. Subtask closed as duplicate to [JDK-8339577](https://bugs.openjdk.org/browse/JDK-8339577)

**Review by azuev-java — APPROVED** [review]

**Pull request closed without merging**